AndroidSDK/src/com/leanplum/internal/LeanplumEventDataManager.java

```diff
@@ -0,0 +1,193 @@
+
+import android.content.ContentValues;
+import android.content.Context;
+import android.database.Cursor;
+
+import android.database.DatabaseErrorHandler;
+import android.database.DatabaseUtils;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+
+import org.json.JSONObject;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * LeanplumEventDataManager class to work with SQLite.
+ *
+ * @author Anna Orlova
+ */
+public class LeanplumEventDataManager {
+  private static final String DATABASE_NAME = "__leanplum.db";
+  private static final int DATABASE_VERSION = 1;
+  private static final String EVENT_TABLE_NAME = "event";
+  private static final String COLUMN_DATA = "data";
+  private static final String KEY_ROWID = "rowid";
+
+  private static SQLiteDatabase database;
+  private static LeanplumDataBaseHelper databaseHelper;
+  private static ContentValues contentValues = new ContentValues();
+
+  public static void init(Context context) {
+    if (database != null) {
+      Log.e("Database is already initialized.");
+      return;
+    }
+    File dbFile = context.getDatabasePath(DATABASE_NAME);
+    boolean needMigration = !dbFile.exists();
+
+    // Create database if needed.
+    try {
+      if(databaseHelper==null) {
+        databaseHelper = new LeanplumDataBaseHelper(context, DATABASE_NAME);
+      }
+      database = databaseHelper.getWritableDatabase();
+    } catch (Throwable t) {
+      Log.e("Cannot create database.", t);
+      Util.handleException(t);
+    }
+
+    // Migrate old data from shared preferences if needed.
+    if (database != null && needMigration) {
+      try {
+        Request.moveOldDataFromSharedPreferences();
+      } catch (Throwable t) {
+        Log.e("Cannot move old data from shared preferences to SQLite table.", t);
+        Util.handleException(t);
+      }
+    }
+  }
+
+  /**
+   * Inserts event to event table.
+   *
+   * @param event String with json of event.
+   */
+  static void insertEvent(String event) {
+    if (database == null) {
+      return;
+    }
+    contentValues.put(COLUMN_DATA, event);
+    try {
+      database.insert(EVENT_TABLE_NAME, null, contentValues);
+    } catch (Throwable t) {
+      Log.e("Unable to insert event to database.", t);
+      Util.handleException(t);
+    }
+  }
+
+  /**
+   * Gets first count events from event table.
+   *
+   * @param count Number of events.
+   * @return List of events.
+   */
+  static List<Map<String, Object>> getEvents(int count) {
+    List<Map<String, Object>> events = new ArrayList<>();
+    if (database == null) {
+      return events;
+    }
+    Cursor cursor = null;
+    try {
+      cursor = database.query(EVENT_TABLE_NAME, new String[] {COLUMN_DATA}, null, null, null,
+          null, KEY_ROWID + " ASC", "" + count);
+      if (cursor.moveToFirst()) {
+        while (!cursor.isAfterLast()) {
```

Comment:
looks like isAfterLast  - Returns whether the cursor is pointing to the position after the last row -  works well here.